File: UIKitExtensions/UIView/UIView+DDExtensions.m
//
//  UIView+DDNibLoading.m
//  UIKitExtensions
//

#import "UIView+DDExtensions.h"

@implementation UIView (DDExtensions)

+ (instancetype)dd_viewWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    NSBundle *bundle = (nibBundleOrNil) ?: [NSBundle mainBundle];
    NSString *name = (nibNameOrNil) ?: NSStringFromClass(self);
    UIView *view = [[bundle loadNibNamed:name owner:self options:nil] firstObject];
    return view;
}

- (UIImageView *)dd_hairlineImageView {
    if ([self isKindOfClass:[UIImageView class]]) {
        if (self.bounds.size.height <= 1.0) {
            return (UIImageView *)self;
        }
    }
    for (UIView *view in self.subviews) {
        UIImageView *hairlineImageView = [view dd_hairlineImageView];
        if (hairlineImageView) {
            return hairlineImageView;
        }
    }
    return nil;
}

@end

File: Sources/UIViewExtension/UIView+DDExtensions.swift
//
//  UIView+DDExtensions.swift
//  
//
//  Created by Anatolij Klimenko on 01.12.2021.
//

import UIKit

extension UIView {
 
    public static func view(with nibName: String = String(describing: self), bundle: Bundle = Bundle.main) -> UIView? {
        bundle.loadNibNamed(nibName, owner: self)?.first as? UIView
    }
    
    public var hairlineImageView: UIImageView? {
        if self.isKind(of: UIImageView.self) {
            if bounds.size.height <= 1.0 {
                return self as? UIImageView
            }
        }
        for view in subviews {
            if let hairline = view.hairlineImageView {
                return hairline
            }
        }
        return nil
    }
}
